packages/platforms/accton/x86-64/as7515-24x/onlp/builds/x86_64_accton_as7515_24x/module/src/psui.c

```diff
@@ -104,50 +88,71 @@ onlp_psui_info_get(onlp_oid_t id, onlp_psu_info_t* info)
     info->hdr.coids[0] = ONLP_THERMAL_ID_CREATE(thermal_count + (pid-1)*NUM_OF_THERMAL_PER_PSU + 1);
     info->hdr.coids[1] = ONLP_THERMAL_ID_CREATE(thermal_count + (pid-1)*NUM_OF_THERMAL_PER_PSU + 2);
 
-    /* Read voltage, current and power */
-    val = 0;
-    ret = psu_pmbus_info_get(pid, "psu_v_in", &val);
-    if (ret == ONLP_STATUS_OK && val) {
-        info->mvin  = val;
-        info->caps |= ONLP_PSU_CAPS_VIN;
-    }
-
-    val = 0;
-    ret = psu_pmbus_info_get(pid, "psu_i_in", &val);
-    if (ret == ONLP_STATUS_OK && val) {
-        info->miin  = val;
-        info->caps |= ONLP_PSU_CAPS_IIN;
-    }
-
-    val = 0;
-    ret = psu_pmbus_info_get(pid, "psu_p_in", &val);
-    if (ret == ONLP_STATUS_OK && val) {
-        info->mpin  = val;
-        info->caps |= ONLP_PSU_CAPS_PIN;
-    }
-
+    /* Get power good status */
     val = 0;
-    ret = psu_pmbus_info_get(pid, "psu_v_out", &val);
-    if (ret == ONLP_STATUS_OK && val) {
-        info->mvout = val;
-        info->caps |= ONLP_PSU_CAPS_VOUT;
+    ret = psu_cpld_status_get(pid, "psu_power_good", &val);
+    if (ret < 0) {
+        info->status |=  ONLP_PSU_STATUS_FAILED;
+        return ONLP_STATUS_E_INTERNAL;
     }
 
-    val = 0;
-    ret = psu_pmbus_info_get(pid, "psu_i_out", &val);
-    if (ret == ONLP_STATUS_OK && val) {
-        info->miout = val;
-        info->caps |= ONLP_PSU_CAPS_IOUT;
+    if (val != PSU_STATUS_POWER_GOOD) {
+        info->status |=  ONLP_PSU_STATUS_UNPLUGGED;
+        info->caps = 0;
```

Comment:
Why do we clear all caps?

Is this AC type correct?
info->caps = ONLP_PSU_CAPS_AC;

Reply:
Yes，this PSU is AC type, "ONLP_PSU_CAPS_AC" need to show in onlpdump.